common/src/com/intellij/plugins/haxe/config/HaxeTarget.java

```diff
@@ -107,4 +108,25 @@ public static void initCombo(@NotNull DefaultComboBoxModel comboBoxModel) {
   public String toString() {
     return description;
   }
+
+  /**
+   * Match the string against the compiler's argument/parameter/flag for the
+   * target output.
+   *
+   * @param compilerTargetArgument - string to compare, e.g. '-js', '-neko'
+   * @return The target matching the flag, or null, if not found.
+   */
+  @Nullable
+  public static HaxeTarget matchOutputTarget(String compilerTargetArgument) {
+    for (HaxeTarget t : HaxeTarget.values()) {
+      if (t.getCompilerFlag().equals(compilerTargetArgument)) {
+        return t;
+      }
+    }
+    // as3 is an old case.
+    if ("-as3".equals(compilerTargetArgument)) {
```

Comment:
in general, I've seem that our AS3 code (in Bravo & pre-Haxe Encore) built by IDEA producing either .swf or .swc file (depending on whether it is an application or a library). 
I'm not 100% (as I didn't test it) whether same applies when compiling from Haxe to AS3 target - though I'd assume it to be generally true ... that's what other developers usually mention when working with IntelliJ and compiling to AS3 target